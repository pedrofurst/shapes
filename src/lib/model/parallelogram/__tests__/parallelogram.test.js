import Parallelogram from '../index';

const newPoint = (x, y) => ({x, y});

const POINT_A = newPoint(1, 5);
const POINT_B = newPoint(2, 3);
const POINT_C = newPoint(3, 6);

const DEFAULT_POINTS = [POINT_A, POINT_B, POINT_C];

const DEFAULT_CENTER_OF_MASS = newPoint(2, 5.5);

const DEFAULT_AREA = 5;

describe('Parallelogram class', () => {

  it('Should instance the class and calculate the last parallelogram point', () => {
    const mockContext = {someContext: 'example'};
    const parallelogram = new Parallelogram(DEFAULT_POINTS, mockContext);
    const lastPoint = parallelogram.points[parallelogram.points.length - 1];
    expect(parallelogram.points.length).toEqual(4);
    expect(parallelogram.points[0]).toEqual(POINT_A);
    expect(parallelogram.points[1]).toEqual(POINT_B);
    expect(parallelogram.points[2]).toEqual(POINT_C);
    expect(lastPoint.x).toEqual(2);
    expect(lastPoint.y).toEqual(8);
    expect(parallelogram.context).toEqual(mockContext);
  });

  it('Should calculate the center of mass', () => {
    const parallelogram = new Parallelogram(DEFAULT_POINTS);
    expect(parallelogram.centerOfMass).toEqual(DEFAULT_CENTER_OF_MASS);
  });

  it('Should calculate the area', () => {
    const parallelogram = new Parallelogram(DEFAULT_POINTS);
    expect(parallelogram.area).toEqual(DEFAULT_AREA);
  });

  it('Should draw the parallelogram', () => {
    const canvas = document.createElement('canvas');
    canvas.setAttribute('width', '200');
    canvas.setAttribute('height', '200');
    const ctx = canvas.getContext('2d');
    const parallelogram = new Parallelogram(DEFAULT_POINTS, ctx);
    parallelogram.draw();
    expect(ctx.__getDrawCalls()).toMatchSnapshot();
  });
});

