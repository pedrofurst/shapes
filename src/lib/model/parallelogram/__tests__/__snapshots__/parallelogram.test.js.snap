// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parallelogram class Should draw the parallelogram 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 1,
            "y": 5,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 1,
            "y": 5,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 2,
            "y": 3,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 3,
            "y": 6,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 2,
            "y": 8,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "closePath",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  Object {
    "props": Object {
      "maxWidth": null,
      "text": "Area: 5",
      "x": 10,
      "y": 30,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillText",
  },
]
`;
